### PR TITLE
Add common forms of generation methods as default

### PR DESCRIPTION
**Pull request by weiyunlu**
*Related Issue(s) or Task(s)*

https://github.com/Zetatango/zetatango/issues/6884

*Why?*

Every model encountered so far uses the same `generate_encryption_epoch` method, and a large number use the same `generate_partition_guid`.  Refactor them as the defaults in the `partition_provider` concern to avoid the need to repeat code and tests for the common cases.

This PR also contains some tests that were in Roadrunner (to be removed there when it's ported to use DaffyLib) that can be made more general to avoid needing to repeat them.

*How?*

Straightforward refactor.

*Risks*

NA

*Requested Reviewers*

List the developers that should review this pull request.

### CODE REVIEW

File: spec/models/child_spec.rb [hunk 1]
```diff
@@ -27,12 +28,66 @@
...
+    it 'sets a guid on create when it is nil' do
+      child = build :child, proxy: proxy, partition_guid: nil
+      child.save!
+      expect(child.send(:partition_guid)).not_to be_blank
+    end
+
+    it 'cannot be changed' do
+      expect do
+        child = create :child, proxy: proxy
+        child.send("#{:partition_guid}=", SecureRandom.base58(16))
```
Comment (gregfletch) [inline spec/models/child_spec.rb]: Codacy Issue found: [Literal interpolation detected.](https://app.codacy.com/gh/Zetatango/daffy_lib/pullRequest?prid=4834584)

File: spec/models/child_spec.rb [hunk 2]
```diff
@@ -27,12 +28,66 @@
...
+    it 'sets a guid on create when it is nil' do
+      child = build :child, proxy: proxy, encryption_epoch: nil
+      child.save!
+      expect(child.send(:encryption_epoch)).not_to be_blank
+    end
+
+    it 'cannot be changed' do
+      expect do
+        child = create :child, proxy: proxy
+        child.send("#{:encryption_epoch}=", SecureRandom.base58(16))
```
Comment (gregfletch) [inline spec/models/child_spec.rb]: Codacy Issue found: [Literal interpolation detected.](https://app.codacy.com/gh/Zetatango/daffy_lib/pullRequest?prid=4834584)

File: spec/models/child_spec.rb (hunk 1 above)
Comment (gregfletch) [inline spec/models/child_spec.rb]: Codacy Issue found: [Literal interpolation detected.](https://app.codacy.com/gh/Zetatango/daffy_lib/pullRequest?prid=4834584)

File: spec/models/child_spec.rb [hunk 3]
```diff
@@ -27,12 +28,66 @@
...
+    it 'sets an encryption epoch on create when it is nil' do
+      child = build :child, proxy: proxy, encryption_epoch: nil
+      child.save!
+      expect(child.send(:encryption_epoch)).not_to be_blank
+    end
+
+    it 'cannot be changed' do
+      expect do
+        child = create :child, proxy: proxy
+        child.send("#{:encryption_epoch}=", SecureRandom.base58(16))
```
Comment (gregfletch) [inline spec/models/child_spec.rb]: Codacy Issue found: [Literal interpolation detected.](https://app.codacy.com/gh/Zetatango/daffy_lib/pullRequest?prid=4834584)

File: lib/daffy_lib/version.rb [hunk 4]
```diff
@@ -1,5 +1,5 @@
 # frozen_string_literal: true
 
 module DaffyLib
-  VERSION = "0.1.0"
+  VERSION = "0.1.1"
```
Comment (gregfletch) [inline lib/daffy_lib/version.rb]: Can we leave version as 0.1.0. We haven't had any releases yet, so no need to rev version number.

Comment (weiyunlu) [reply lib/daffy_lib/version.rb]: I did push it to RubyGems actually: https://rubygems.org/gems/daffy_lib

**Comment by weiyunlu** [comment]
> Looks good. Why is the gem being added?

Not sure what you mean by this?

**Pull request merged**